### DISCUSSION TITLE
Basic Setup Guide

### DISCUSSION THREAD

**Discussion by mgross9** [umbraco/Umbraco-CMS#16085] [Features and ideas]
Hello, I am relatively new to the Umbraco community. I am trying to build the cms from the code here (from the release 13.3.0 branch, which I think is the release for the most recent umbraco version), but I am having trouble building the actual project. I am using Visual Studio 2022 to build the project. When I open the umbraco.sln file in Visual Studio, I am immediately met with about 136 "errors", and I am unable to build any of the projects.

Some of the errors I am running into are as follows:
1. The type or namespace name 'AspNetCore' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?)
This shows up with any using statement that references 'Microsoft.AspNetCore' or any in its sub namespaces. This seems like an outdated reference as it is depreciated and considered legacy.

2. The type or namespace name 'Logging' does not exist in the namespace 'Umbraco.Cms.Core' (are you missing an assembly reference?) This error appears for more than just Logging, as multiple namespaces in the 'Core' project are also claiming to not exist (i.e Models, Routing, Scoping, Security, Services, Web). Notably though, the 'Cache' namespace does exist and can be found. The namespaces do not appear to be actually missing, just somehow misreferenced.

3. The "GetUmbracoBuildVersion" task was not given a value for the required parameter "GitCommitIdShort". 
This error I do not understand. Is there a way to set a value for GitCommitIdShort?	

I think in general the project could use a "Basic Installation and Getting Started" guide that includes a step-by-step guide on how to set up the project after downloading from Github. I have been trying for a few days just to build the project to start work with it, but have been unsuccessful so far. If a guide like this already exists, can someone reply with a link to it?

**Comment by jossgowans** [comment]
Hi,
if you want to start building websites in Umbraco, you don't need to build from the source code. I made that mistake when I first looked at Umbraco many years ago. If building websites is what you want to do, I'd recommend going to [https://psw.codeshare.co.uk/](https://psw.codeshare.co.uk/) clicking on the Install Script tab, copying that script and running it in a terminal window. Make sure you have .Net 8 installed first.

If you have you found a bug or want to make a change the the CMS, or understand the inner workings, there is a link in the read me section on the landing page of the project her to [contributing ](https://github.com/umbraco/Umbraco-CMS/blob/contrib/.github/CONTRIBUTING.md) which explains that you should use the _contrib_ branch and there is also a link from there to how to [build the solution](https://github.com/umbraco/Umbraco-CMS/blob/contrib/.github/BUILD.md) locally.

For general questions I'd recommend the [Umbraco Discord channel ](https://discord.com/channels/869656431308189746/) - Sorry, I'm note sure if that's the correct link 
and also the forum on [Our Umbraco](https://our.umbraco.com/forum)

It's also worth watching Paul Seal's video series. This is for Version 10 and above, so don't be put of by the reference to v10 https://www.youtube.com/watch?v=qEZtjxDEK8g

All the best in your Umbraco journey
tom

**Reply by jossgowans** [reply]
Sorry, logged in with different Github account there, I'm Tom Madden TwoMoreThings__
